File: src/ui/src/main/java/com/amazon/sample/ui/services/checkout/model/ShippingOption.java
package com.amazon.sample.ui.services.checkout.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ShippingOption {
    private String name;

    private Integer amount;

    private String token;

    private Integer estimatedDays;
}

File: src/checkout/src/checkout/models/ShippingOption.ts
import { IsInt, IsNumber, IsString } from 'class-validator';

export class ShippingOption {

  @IsString()
  name : string;

  @IsInt()
  amount : number;

  @IsString()
  token : string;

  @IsInt()
  estimatedDays : number;

}